pandas/io/formats/format.py

```diff
@@ -1514,18 +1514,25 @@ def __init__(
 
     def _format_strings(self) -> List[str]:
         """ we by definition have DO NOT have a TZ """
-        values = self.values
-
-        if not isinstance(values, DatetimeIndex):
-            values = DatetimeIndex(values)
+        values = np.asarray(self.values)
+        flat_values = values.ravel() if values.ndim > 1 else values
+        flat_values = DatetimeArray(flat_values)
 
         if self.formatter is not None and callable(self.formatter):
-            return [self.formatter(x) for x in values]
+            fmt_values = [self.formatter(x) for x in flat_values]
+        else:
```

Comment:
why are we not using just the _format_native_types here? this makes this way more complicated.

Reply:
Sorry, which line are you referring to? Did you mean instead of calling `self.formatter`?

Reply:
I think the issue is that self.formatter may be user-supplied at some point in the process?